internal/weldrcheck/blueprints_test.go

```diff
@@ -0,0 +1,754 @@
+// Package weldrcheck - blueprints contains functions to check the blueprints API
+// Copyright (C) 2020 by Red Hat, Inc.
+
+// Tests should be self-contained and not depend on the state of the server
+// They should use their own blueprints, not the default blueprints
+// They should not assume version numbers for packages will match
+// They should run tests that depend on previous results from the same function
+// not from other functions.
+// The blueprint version number may get bumped if the server has had tests run before
+// do not assume the bp version will match unless first deleting the old one.
+
+// +build integration
+
+package weldrcheck
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	//	"strings"
+	"testing"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/osbuild-composer/internal/client"
+)
+
+// Hold test state to share between tests
+var testState *TestState
```

Comment:
The comment  here triggered my spider sense. There shouldn't be any state shared between the tests (as much as possible) and in fact from what I can see only the socket is shared.  Maybe this just needs better wording. OTOH if we need only the socket why not make a global variable called `testSocket` or similar ?

Reply:
So the alternative is to setup and tear down the connection in each test, which is slower and lots more code. There is no way to pass this from the main test function to the test functions so it has to be a global, and it has to be in each of the test files. It also contains the API version so that the tests can be extended to run API v1 specific tests at some point soon.

I also want to share this setup/state code with other tests (there are going to be sources_test.go, compose_test.go, etc. at some point).